lib/demangler-win32.js

```diff
@@ -0,0 +1,163 @@
+                        logger.warn("something went wrong: ", label, " doesn't have an undecoration.");
+                    }
+                }
+                asmLine.text = beforeComment + (afterComment ? afterComment : "");
+            }
+        }
+
+        return this.result;
+    }
+
+    execDemangler() {
+        let translations = {};
+
+        const demangleSingleSet = names => {
+            const args = [this.flags, ... names];
+            return this.compiler.exec(this.demanglerExe, args, this.compiler.getDefaultExecOptions())
+                .then(output => {
+                    const outputArray = utils.splitLines(output.stdout);
+
+                    for (let i = 0; i < outputArray.length; ++i) {
+                        let tmp = outputArray[i].match(/^Undecoration of :- "(.*)"/);
+                        if (tmp) {
+                            const decoratedName = tmp[1];
+                            ++i;
+                            tmp = outputArray[i].match(/^is :- "(.*)"/);
+                            if (!tmp) {
+                                logger.error("Broken undname: ", outputArray[i - 1], outputArray[i]);
+                            } else {
+                                if (this.hasQuotesAroundDecoratedLabels) {
+                                    translations['"' + decoratedName + '"'] = tmp[1];
+                                } else {
+                                    translations[decoratedName] = tmp[1];
+                                }
+                            }
+                        }
+                    }
+                });
+        };
+
+        // give some space for `undname` as well as the flag
+        // should probably be done more correctly
+        const maxCommandLineLength = 8000;
+        const commandLineArray = [];
+        this.win32RawSymbols.sort();
+
+        let lastSymbol = null;
+        let currentLength = 0;
+        let currentArray = [];
+        for (const symb of this.win32RawSymbols) {
+            if (symb === lastSymbol) {
+                continue;
+            }
+            lastSymbol = symb;
+            // note: plus one for the space after an argument
+            if (currentLength + symb.length + 1 < maxCommandLineLength) {
+                currentArray.push(symb);
+                currentLength += symb.length + 1;
+            } else {
+                commandLineArray.push(currentArray);
+                currentLength = symb.length + 1;
+                currentArray = [symb];
+            }
+        }
+        if (currentArray.length !== 0) {
+            commandLineArray.push(currentArray);
+        }
+
+        const demangleAllArgs = commandLineArray.map(demangleSingleSet);
+        return Promise.all(demangleAllArgs).then(() => translations);
+    }
+
+    process(result) {
+        if (!this.demanglerExe) {
+            logger.error("Attempted to demangle, but there's no demangler set");
+            return result;
```

Comment:
this should return a Promise, though it shouldn't happen in most cases anyway

Reply:
ooh, good point.